FRAuth/FRAuth/DeviceBinding/ApplicationPinDeviceAuthenticatorDelegate.swift

```diff
@@ -0,0 +1,72 @@
+// 
+//  ApplicationPinDeviceAuthenticatorDelegate.swift
+//  FRAuth
+//
+//  Copyright (c) 2023 ForgeRock. All rights reserved.
+//
+//  This software may be modified and distributed under the terms
+//  of the MIT license. See the LICENSE file for details.
+//
+
+
+import UIKit
+
+
+/// ApplicationPinDeviceAuthenticatorDelegate for collecting the Pin
+public protocol ApplicationPinDeviceAuthenticatorDelegate: AnyObject {
```

Comment:
The approach seems different with Android, instead of setting a delegate, Android allows developer to override the existing ApplicationPinDeviceAuthenticator.requestForCredentials() to collect the pin. I am ok with both approaches, if we want to use a delegate (or interface), Android can easily change as well. Lets discuss.